### ISSUE TITLE
TimeRange isBefore and isAfter behave the same

### ISSUE THREAD

**Issue by nemo83** [Hyperion-BT/helios#73]
Hi,

Just noticed that the `is_before` and `is_after` on `TimeRange` seem to behave the same.

**Comment by christianschmitz** [comment]
I've verified that the unit tests are correct.

I'll add a `TimeRange.show()` method, so you can debug better this issue:

Will return a string in form of `"[-inf,+inf]"`, `"[+inf,-inf]"`,`"[-inf,16187238172]"`, `"[16187238172, +inf]"` or `"[16187238172, 16188000000]"`. Note: `"[+inf,-inf]"` is equivalent to `TimeRange::NEVER`

**Comment by christianschmitz** [comment]
v0.13.7 now has TimeRange.show()

**Comment by gavinharris-dev** [comment]
I get IR empty list when I forget to supply a TX valid From and To in the transaction. 
I’ve also had ‘issues’ with validating a start date on a TX. If you make the TX::validFrom a little earlier than current date, you could end up before the “start time” set in the datum. I needed to put the Tax valid from to be a little earlier as the Cardano Nodes were rejecting the TX if it was exactly Current Date.  


sorry on the phone, can give a better explanation later

**Comment by christianschmitz** [comment]
@gavinharris-dev There is no default validFromSlot, it's just empty.

How about adding methods: `tx.validFromSlot` and `tx.validToSlot` ?

**Comment by gavinharris-dev** [comment]
I’ll need to look again, just out and about now!
Thinking about it I think I have a bug in my contract, if I have to subtract a couple of mins from the validFromSlot then I should add it back before validating against my “start date”.

**Comment by nemo83** [comment]
This is now fixed, mis-use of the time-range. I forgot to supply `validFrom/validTo`. Added them to my offchain code and looking good.

IIRC if I used the `tx.time_range.is_after` it **would** not fail. It would throw an IR error if `start/end` was used instead. I'm only getting familiar with the `time_range` now, so not sure what would be the best behaviour, but maybe errors can be improved?

For me the issue can be closed, but given you're talking about something else I might leave this one open and let you guys close it.

Thanks again for the support.
